### PR TITLE
Fixed all build processes.

### PR DESCRIPTION
**Pull request by timothygwebb**
Fixed all build processes.

### CODE REVIEW

**Comment by timothygwebb** [comment]
@fordnox @evrifaessa  This pull request integrates all other pull requests and fixes the build issues. I have tagged the security issue with this pull request.

**Comment by evrifaessa** [comment]
Trying with PHP 8, the installer doesn't pass the database process with this error:

`[Mon Apr 19 09:02:28.059870 2021] [php:error] [pid 71876:tid 1832] [client ::1:18736] PHP Fatal error:  Uncaught Error: Undefined constant "BB_PATH_LANGS" in C:\\xampp\\htdocs\\bb-library\\Box\\Translate.php:78\nStack trace:\n#0 C:\\xampp\\htdocs\\bb-di.php(358): Box_Translate->setup()\n#1 C:\\xampp\\htdocs\\install\\install.php(410): Box_Installer->{closure}()\n#2 C:\\xampp\\htdocs\\install\\install.php(109): Box_Installer->generateEmailTemplates()\n#3 C:\\xampp\\htdocs\\install\\install.php(418): Box_Installer->run('install')\n#4 {main}\n  thrown in C:\\xampp\\htdocs\\bb-library\\Box\\Translate.php on line 78, referer: http://localhost/install/install.php`

As I can't get through the installation process, I can't test further. Let me try again with PHP 7.

File: tests/bb-modules/Servicehosting/ServiceTest.php [hunk 1]
```diff
@@ -163,7 +163,7 @@ public function testaction_renew()
         $orderModel = new \Model_ClientOrder();
         $orderModel->loadBean(new \RedBeanPHP\OODBBean());
 
-        $model = new \Model_ServiceHosting();
+        $model = new \Model_ServiceHostingHP();
```
Comment (evrifaessa) [inline tests/bb-modules/Servicehosting/ServiceTest.php:166]: Uppercase HP

File: tests/bb-modules/Servicehosting/ServiceTest.php [hunk 2]
```diff
@@ -784,7 +784,7 @@ public function testtoApiArray()
         $hostingServer = new \Model_ServiceHostingServer();
         $hostingServer->loadBean(new \RedBeanPHP\OODBBean());
         $hostingServer->manager = 'Custom';
-        $hostingHp = new\Model_ServiceHostingHp();
+        $hostingHp = new \Model_ServiceHostingHp();
```
Comment (evrifaessa) [inline tests/bb-modules/Servicehosting/ServiceTest.php:787]: Lowercase Hp, which one is correct?

Comment (timothygwebb) [reply tests/bb-modules/Servicehosting/ServiceTest.php:787]: Lowercase Hp is correct.

File: tests/bb-modules/Servicehosting/ServiceTest.php (hunk 1 above)
Comment (evrifaessa) [reply tests/bb-modules/Servicehosting/ServiceTest.php]: Fixed with f4845e5eabd4362efc14bbb4636610ef0add11bf

**Comment by evrifaessa** [comment]
PHP 8 is still not supported. But at least it doesn't seem to be erroring out in tests. Merging.

**Comment by evrifaessa** [comment]
@timothygwebb something similar happened here:
https://travis-ci.org/github/boxbilling/boxbilling/builds/767548724

mind fixing?
